### PR TITLE
Expose additional CloudFoundry metadata in the DCA API

### PR DESCRIPTION
**Pull request by NouemanKHAL**
### What does this PR do?
Exposes some CF Cloud Controller API data (CF Orgs, CF Org Quotas and preprocessed CF Applications) in the Cluster Agent API.

### Motivation

Improve the performance of the PCF nozzles and reduce their load on the Cloud Controller API in large PCF environments.

### Describe how to test/QA your changes

* Upload the cluster agent to a PCF environment.
* Query the endpoints and verify the integrity of the data.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ahmed-mez — APPROVED** [review]
Changes in `cmd/cluster-agent/api/server.go` look good

File: pkg/util/cloudproviders/cloudfoundry/cccache.go [hunk 1]
```diff
@@ -30,29 +31,55 @@ type CCCacheI interface {
...
+	GetOrg(string) (*cfclient.V3Organization, error)
+
+	// GetOrgs returns all orgs in the cache
+	GetOrgs() ([]*cfclient.V3Organization, error)
+
+	// GetOrgQuotas returns all orgs quotas in the cache
+	GetOrgQuotas() ([]*CFOrgQuota, error)
+
+	// GetProcesses returns all processes of the given appGUID in the cache
+	GetProcesses(string) ([]*cfclient.Process, error)
```
Comment (therve) [inline pkg/util/cloudproviders/cloudfoundry/cccache.go:50]: I don't think you use that one. We should probably remove it if it's unused.

**Review by therve — APPROVED** [review]
Minor comment otherwise looks great.

**Review by vickenty — APPROVED** [review]

File: releasenotes/notes/expose-pcf-nozzle-data-in-dca-api-dd544b82ac8a744a.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    Expose additional CloudFoundry metadata in the DCA API to be used
+    by PCF firehose nozzles in order to reduce the load on the CC API.
```
Comment (alai97) [inline releasenotes/notes/expose-pcf-nozzle-data-in-dca-api-dd544b82ac8a744a.yaml:12]:
```suggestion
    Expose additional CloudFoundry metadata in the DCA API that the PCF firehose nozzles
    use to reduce the load on the CC API.
```

**Review by alai97 — APPROVED** [review]
Looks good for release notes!

**Pull request merged**